buffer/src/main/java/io/netty/buffer/PoolThreadCache.java

```diff
@@ -179,6 +179,9 @@ private boolean allocate(MemoryRegionCache<?> cache, PooledByteBuf buf, int reqC
     @SuppressWarnings({ "unchecked", "rawtypes" })
     boolean add(PoolArena<?> area, PoolChunk chunk, ByteBuffer nioBuffer,
                 long handle, int normCapacity, SizeClass sizeClass) {
+        if (freed.get()) {
+            return false;
+        }
```

Comment:
should we move this closer to when we actually add it to the data-structure ?

Reply:
Fine, that's will be better.

Reply:
If we check it in MemoryRegionCache.add, the class `MemoryRegionCache` is static, we should change it to a non-static class. I'm not sure if is it deserved.

Reply:
Just put the check in front of `return cache.add(chunk, nioBuffer, handle, normCapacity);`

Reply:
done